DEVPROD-5859 Remove commit queue docs

bsamek opened the pull request:
DEVPROD-5859

### Description

This removes docs for the legacy commit queue. All projects have migrated. I
will not merge this until we remove the UI allowing users to select the commit
queue https://github.com/evergreen-ci/ui/pull/178. I'm waiting on one user to
confirm for me that all looks good, but wanted to prepare this PR.

### Testing

N/A, docs only.

### Documentation

This!

Kimchelly reviewed (changes requested):
Sorry, GitHub doesn't let me comment on unchanged lines, so all I can do is list out the comments...

* Is the [commit queue alias example](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Project-and-Distro-Settings.md?plain=1#L674-L680) still valid?
* [The commit queue expansion](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Project-Configuration-Files.md?plain=1#L796) should say it's true for the merge queue now.
* Is [the commit queue requester](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Project-Configuration-Files.md?plain=1#L799) still valid, or should it be changed to `github_merge_queue`?
* Is [the task priority for merge queue tasks](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Task-Priority.md?plain=1#L14) supposed to be changed to merge queue? Do merge queue tasks get bonus weight in the scheduler?
* Should [this thing about parameters and the commit queue](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Parameterized-Builds.md?plain=1#L39) be removed/edited?
* [`merge_test`](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Evergreen-Data-for-Analytics.md?plain=1#L30) needs to be removed from the Trino docs.
* Are we going to rename the "GitHub & Commit Queue" section in the UI, or is the commit queue part going to stay?

bsamek commented on docs/Project-Configuration/Github-Integrations.md:
```diff
@@ -81,19 +79,6 @@ Sometimes Evergreen has trouble sending updated GitHub statuses, so the checks o
 
 GitHub only allows one set of statuses for every commit SHA. If you have a situation where you have two PRs where the HEAD commit is the same, Evergreen will only create a patch for the first one and will make a comment on the second one explaining why. In general, if your tasks do not require the context of the PR (most notably the branch name, which is known to be used in some s3.put tasks), th ...
 
-## Commit Queue
```
Added.

bsamek commented:
> Sorry, GitHub doesn't let me comment on unchanged lines, so all I can do is list out the comments...
> 
> * Is the [commit queue alias example](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Project-and-Distro-Settings.md?plain=1#L674-L680) still valid?

Yes. I added some clarifying text.

> * [The commit queue expansion](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Project-Configuration-Files.md?plain=1#L796) should say it's true for the merge queue now.

Fixed.

> * Is [the commit queue requester](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Project-Configuration-Files.md?plain=1#L799) still valid, or should it be changed to `github_merge_queue`?

Fixed.

> * Is [the task priority for merge queue tasks](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Task-Priority.md?plain=1#L14) supposed to be changed to merge queue? Do merge queue tasks get bonus weight in the scheduler?

They do. Fixed.

> * Should [this thing about parameters and the commit queue](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Parameterized-Builds.md?plain=1#L39) be removed/edited?

Removed.

> * [`merge_test`](https://github.com/evergreen-ci/evergreen/blob/aab5ad0615f1dbeb7d0978fbd58f59ea58081a3a/docs/Project-Configuration/Evergreen-Data-for-Analytics.md?plain=1#L30) needs to be removed from the Trino docs.

I left that in on the theory that people might still review historical tasks.

> * Are we going to rename the "GitHub & Commit Queue" section in the UI, or is the commit queue part going to stay?

I plan to rename this.

Kimchelly reviewed (approved):
LGTM mod could we mention that `merge_test` in the Trino docs is deprecated? I believe Trino data also has a TTL, so it'll eventually never appear there either.

Also, so excited to delete this 💣

ybrill commented:
Drive-by comment (I saw the commit and was curious): do we want to remove the images (like [this one](https://github.com/evergreen-ci/evergreen/blob/0a8c61fba1bfe268bda561d9c1537ee4e77f6017/docs/images/merge_task.png))? Or are they still being referenced somewhere?

bsamek commented:
> Drive-by comment (I saw the commit and was curious): do we want to remove the images (like [this one](https://github.com/evergreen-ci/evergreen/blob/0a8c61fba1bfe268bda561d9c1537ee4e77f6017/docs/images/merge_task.png))? Or are they still being referenced somewhere?

Ah they should, thanks! Opened https://github.com/evergreen-ci/evergreen/pull/8015
